### PR TITLE
Allowing config flag to turn off javascript controls

### PR DESCRIPTION
**Pull request by mistercrunch**
screen shot 2018-02-10 at 5 42 28 pm

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -948,7 +956,12 @@ def get_form_data(self):
 
         if request.args.get('viz_type'):
             # Converting old URLs
-            d = cast_form_data(request.args)
+            d = cast_form_data(d)
+
+        for k in FORM_DATA_KEY_BLACKLIST:
```
Comment (betodealmeida) [inline superset/views/core.py]: Nit: you can do this in a single line

```python
d = {k: v for k, v in d.items() if k not in FORM_DATA_KEY_BLACKLIST}
```

**Review by betodealmeida — APPROVED** [review]
Looks good, one question only: have you considered completely removing the form in this case, instead of having the warning? Can we also disable the Javascript form in this case, so it's read only?

File: superset/views/core.py (hunk 1 above)
Comment (mistercrunch) [reply superset/views/core.py]: extra points for avoiding the mutation

**Comment by mistercrunch** [comment]
Since the default is to set the feature off (secure should be the default), I want for the users to see that the feature exists otherwise there's no way for the users to discover that the feature exists.

I looked quickly at the react-ace to see if a `disabled` prop exists but apparently not, so decided to just have the warning icon instead:
https://github.com/securingsincity/react-ace/blob/master/src/ace.js

**Comment by mistercrunch** [comment]
Oh looks like there's a readOnly prop I missed on my first pass

**Comment by mistercrunch** [comment]
@betodealmeida made the fields `readOnly`

**Comment by mistercrunch** [comment]
For the record a security expert mentioned in an email:
```
If you create an iframe in the `none` origin, you could then use the postMessage API to send it the function definition (as a string) along with the data. It could then eval the function definition, run it on the data, and then use the postMessage API to send the mutated data back to the app to be rendered. 
```

**Comment by mistercrunch** [comment]
Addressed comments, merging.

**Pull request merged**